tests/e2e/upgrade_test.go

```diff
@@ -95,6 +96,34 @@ var _ = Describe("Upgrade", Label(tests.LabelUpgrade, tests.LabelNoOpenshift), O
 		if testLevelEnv.Depth < int(level) {
 			Skip("Test depth is lower than the amount requested for this test")
 		}
+
+		err := env.DeleteNamespace(operatorNamespace)
+		if !apierrs.IsNotFound(err) {
+			Expect(err).NotTo(HaveOccurred())
+		}
+
+		err = env.CreateNamespace(operatorNamespace)
+		Expect(err).NotTo(HaveOccurred())
+
+		// As the 'cnpg-system' namespace is deleted after each case is completed
+		// we should create the namespace again before each case and create the
+		// create pull image secret again
+		dockerServer := os.Getenv("DOCKER_SERVER")
+		dockerUsername := os.Getenv("DOCKER_USERNAME")
+		dockerPassword := os.Getenv("DOCKER_PASSWORD")
+		if dockerServer != "" && dockerUsername != "" && dockerPassword != "" {
+			_, _, err := testsUtils.Run(fmt.Sprintf(`kubectl -n %v create secret docker-registry
+			cnpg-pull-secret
+			--docker-server="%v"
+			--docker-username="%v"
+			--docker-password="%v"`,
+				operatorNamespace,
+				dockerServer,
+				dockerUsername,
+				dockerPassword,
+			))
+			Expect(err).NotTo(HaveOccurred())
+		}
```

Comment:
isn't this covered in `ensure_image_pull_secret` in the run-e2e.sh script?
Why is it here?

Reply:
please look at https://github.com/cloudnative-pg/cloudnative-pg/pull/3624#issuecomment-1875083582